src/averageAges.js

```diff
@@ -15,11 +15,18 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => (
+    person.sex === 'm'
+    && (century
+      ? century === Math.ceil(person.died / 100)
+      : true)
+  ));
+
+  const menAges = men.map(person => person.died - person.born);
+
+  const totalAge = menAges.reduce((age, average) => age + average, 0);
```

Comment:
You use reduce in every function. It should be added to the `calculateAverageAge` function.